packages/firestore/src/local/memory_persistence.ts

```diff
@@ -119,3 +154,154 @@ export class MemoryPersistence implements Persistence {
 export class MemoryTransaction implements PersistenceTransaction {
   constructor(readonly currentSequenceNumber: ListenSequenceNumber) {}
 }
+
+export class MemoryLruDelegate implements ReferenceDelegate, LruDelegate {
+  private additionalReferences: ReferenceSet | null;
+  private orphanedSequenceNumbers: ObjectMap<
+    DocumentKey,
+    ListenSequenceNumber
+  > = new ObjectMap(k => encode(k.path));
+
+  readonly garbageCollector: LruGarbageCollector;
+
+  constructor(private readonly persistence: MemoryPersistence) {
+    this.garbageCollector = new LruGarbageCollector(this);
+  }
+
+  forEachTarget(
+    txn: PersistenceTransaction,
+    f: (q: QueryData) => void
+  ): PersistencePromise<void> {
+    return this.persistence.getQueryCache().forEachTarget(txn, f);
+  }
+
+  getTargetCount(txn: PersistenceTransaction): PersistencePromise<number> {
+    return this.persistence.getQueryCache().getTargetCount(txn);
+  }
+
+  forEachOrphanedDocumentSequenceNumber(
+    txn: PersistenceTransaction,
+    f: (sequenceNumber: ListenSequenceNumber) => void
+  ): PersistencePromise<void> {
+    this.orphanedSequenceNumbers.forEach((_, sequenceNumber) =>
+      f(sequenceNumber)
+    );
+    return PersistencePromise.resolve();
+  }
+
+  setInMemoryPins(inMemoryPins: ReferenceSet): void {
```

Comment:
I much prefer the name on Android (setAdditionalReferences). It conveys meaning even to those folks that don't know that Lru "pins" documents.

Reply:
It has already been changed on Android to `inMemoryPins` as part of the indexeddb portion of this port.

Reply:
Hm. Sad panda. Reminder for myself to never to go on parental leave.